auth_backends/tests/test_backends.py

```diff
@@ -255,16 +257,49 @@ def test_partial_pipeline(self):
         self.do_partial_pipeline()
 
     def test_logout_url(self):
-        """ The property should return the provider's logout URL. """
+        """
+        Verify the property returns the provider's logout URL.
+        """
+        logout_url_without_query_params = '{}/logout'.format(self.url_root)
+
         self.assertEqual(
             self.backend.logout_url,
-            '{}/logout?client_id={}&redirect_url={}'.format(
-                self.url_root,
-                self.client_key,
-                self.logout_redirect_url,
-            )
+            logout_url_without_query_params,
+        )
+
+        # NOTE (CCB): We use the strategy's method, rather than override_settings, because the TestStrategy
+        # class being used does not rely on Django settings.
+        self.strategy.set_settings({'SOCIAL_AUTH_{}_LOGOUT_REDIRECT_URL'.format(self.name): self.logout_redirect_url})
+
+        expected_query_params = '?client_id={}&redirect_url={}'.format(self.client_key, self.logout_redirect_url)
+
+        self.assertEqual(
+            self.backend.logout_url,
+            logout_url_without_query_params + expected_query_params,
         )
 
+    def test_authorization_url(self):
+        """
+        Verify the method utilizes the public URL, if one is set.
+        """
+        authorize_location = '/oauth2/authorize'
+        self.assertEqual(self.backend.authorization_url(), self.url_root + authorize_location)
+
+        expected = 'http://public.example.com'
+        # NOTE (CCB): We use the strategy's method, rather than override_settings, because the TestStrategy
+        # class being used does not rely on Django settings.
+        self.strategy.set_settings({'SOCIAL_AUTH_{}_PUBLIC_URL_ROOT'.format(self.name): expected})
+        self.assertEqual(self.backend.authorization_url(), expected + authorize_location)
+
     def test_end_session_url(self):
-        """ The method should return the provider's logout URL. """
-        self.assertEqual(self.backend.end_session_url(), '{}/logout'.format(self.url_root))
+        """
+        Verify the method returns the provider's logout URL (sans any redirect URLs in the query parameters).
+        """
+        logout_location = '/logout'
+        self.assertEqual(self.backend.end_session_url(), self.url_root + logout_location)
+
+        expected = 'http://public.example.com'
+        # NOTE (CCB): We use the strategy's method, rather than override_settings, because the TestStrategy
+        # class being used does not rely on Django settings.
+        self.strategy.set_settings({'SOCIAL_AUTH_{}_PUBLIC_URL_ROOT'.format(self.name): expected})
```

Comment:
Consider adding a helper method for calling `self.strategy.set_settings` so the comment noted above it doesn't need to be repeated.